src/python/grpcio/grpc/_server.py

```diff
@@ -958,11 +958,27 @@ def add_generic_rpc_handlers(self, generic_rpc_handlers):
         _add_generic_handlers(self._state, generic_rpc_handlers)
 
     def add_insecure_port(self, address):
-        return _add_insecure_port(self._state, _common.encode(address))
+        port = _add_insecure_port(self._state, _common.encode(address))
+        if port == 0:
+            # The Core API doesn't return a failure message. The best we can do
+            # is raising an exception to prevent further confusion.
+            raise RuntimeError('Failed to bind to address %s; set '
+                               'GRPC_VERBOSITY=debug env to see detailed error '
+                               'message.' % address)
+        else:
+            return port
 
     def add_secure_port(self, address, server_credentials):
-        return _add_secure_port(self._state, _common.encode(address),
+        port = _add_secure_port(self._state, _common.encode(address),
                                 server_credentials)
+        if port == 0:
+            # The Core API doesn't return a failure message. The best we can do
+            # is raising an exception to prevent further confusion.
+            raise RuntimeError('Failed to bind to address %s; set '
```

Comment:
This logic is repeated four times. It should probably be pulled out to a common validation function.

Reply:
Abstracted out to `_common.validate_port_binding_result`. Good idea!